### PR TITLE
LCAM-221|Migrate the Business Rules From CROWN_COURT.DETERMINE_CROWN_…

### PR DESCRIPTION
**Pull request by venkat980**
…REPORDER_DATE

## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-221)

Migrate the Business Rules From CROWN_COURT.DETERMINE_CROWN_REPORDER_DATE.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by venkat980** [comment]
I have left the stored procedure code as a reference for the review, I will remove this after the review that would get rid of the code smells.

File: crown-court-proceeding/src/main/resources/application.yaml [hunk 1]
```diff
@@ -90,3 +90,10 @@ cloud-platform:
 crown-court-proceeding:
   security:
     issuer-uri: http://localhost:${server.port}
+
+maatApi:
+  oAuthEnabled: true
+  baseUrl: ${MAAT_API_BASE_URL}
+  assessments-domain: /api/internal/v1/assessment
+  ioj-appeal-endpoints:
+    find-url: ${maatApi.assessments-domain}/ioj-appeal/repId/{repId}/current-passed
```
Comment (muthusubra) [inline crown-court-proceeding/src/main/resources/application.yaml:99]: Change this to Query Param

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/RepOrderService.java [hunk 2]
```diff
@@ -165,4 +167,96 @@ public void determineRepTypeByDecisionReason(CrownCourtsActionsRequestDTO reques
...
+                          p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_REPORDER_DATE := p_app_obj.committal_date;
+                       END IF;
+                    END IF;
+                */
+                case EITHER_WAY:
+                    if (requestDTO.getMagCourtOutcome() == MagCourtOutcome.COMMITTED_FOR_TRIAL) {
+                        switch (requestDTO.getDecisionReason()) {
+                            case GRANTED -> crownCourtSummary.setRepOrderDate(requestDTO.getDecisionDate());
+                            case FAILIOJ, FAILMEIOJ, FAILMEANS -> crownCourtSummary.setRepOrderDate(requestDTO.getCommittalDate());
+                            default -> crownCourtSummary.setRepOrderDate(null);
```
[the remark this answers is not in the document]
Comment (venkat980) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/RepOrderService.java:209]: Sonarlint was suggesting to add a default condition.

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/RepOrderService.java [hunk 3]
```diff
@@ -165,4 +167,96 @@ public void determineRepTypeByDecisionReason(CrownCourtsActionsRequestDTO reques
...
+                    THEN
+                       p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_REPORDER_DATE := p_app_obj.date_received;
+                    end;
+                    */
+                case APPEAL_CC:
+                    try {
+                        IOJAppealDTO iojAppealDTO = maatCourtDataService
+                                .getCurrentPassedIOJAppealFromRepId(requestDTO.getRepId(), requestDTO.getLaaTransactionId());
+                        crownCourtSummary.setRepOrderDate(iojAppealDTO.getDecisionDate());
+                    } catch (Exception ex) {
```
[the remark this answers is not in the document]
Comment (venkat980) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/RepOrderService.java:241]: Agree. I have created LCAM-322 to enhance the current webclient handling.

File: crown-court-proceeding/src/main/resources/application.yaml (hunk 1 above)
Comment (venkat980) [reply crown-court-proceeding/src/main/resources/application.yaml:99]: LCAM-322 is created to support query params handling in the webclient, also waiting on the outcome of GraphQL implementation spike.

**Review by muthusubra — APPROVED** [review]

**Review by muthusubra — APPROVED** [review]

**Pull request merged**